internal/backends/collection.go

```diff
@@ -96,6 +99,9 @@ type QueryResult struct {
 // It also can be used to close the returned iterator and free underlying resources,
 // but doing so is not necessary - the handler will do that anyway.
 //
+// If DisablePushdown is true, no pushdown will be performed.
+// In such case Filter, Limit, and Sort cannot be set.
```

Comment:
Essentially, you are having a function `func foo(a *A, b bool)` that says something like "if b is `true`, `foo` behaves as if `a` was `nil`". But then the caller could just pass `nil` to `a`. Parameter `b` is not needed.

So please remove `DisablePushdown` from the backend and arrange the handler to call backend methods with correct parameters.